packages/grafana-ui/src/components/DataSourceSettings/DataSourceHttpSettings.tsx

```diff
@@ -146,24 +148,20 @@ export const DataSourceHttpSettings = (props: HttpSettingsProps) => {
   const azureAuthEnabled: boolean =
     (azureAuthSettings?.azureAuthSupported && azureAuthSettings.getAzureAuthEnabled(dataSourceConfig)) || false;
 
-  const connectionLabel = connectionElements?.label ? connectionElements?.label : 'URL';
-
   return (
     <div className="gf-form-group">
       <>
         <h3 className="page-heading">HTTP</h3>
         <div className="gf-form-group">
-          {defaultUrl && (
-            <div className="gf-form">
-              <FormField
-                interactive={connectionElements?.tooltip ? true : false}
-                label={connectionLabel}
-                labelWidth={13}
-                tooltip={urlTooltip}
-                inputEl={urlInput}
-              />
-            </div>
-          )}
+          <div className="gf-form">
+            <FormField
+              interactive={urlDocs ? true : false}
+              label={urlLabel ?? 'URL'}
+              labelWidth={13}
+              tooltip={urlTooltip}
+              inputEl={urlInput}
+            />
+          </div>
```

Comment:
I reverted this defaultUrl being optional, this whole component does not work without this option here. An alternative would be a property with "urlField: React.ReactNode"  so callers could insert a custom field here. But not showing the url property under this HTTP option category would be pretty confusing :)

Reply:
This makes sense!